commands/deploy.go

```diff
@@ -187,34 +220,21 @@ func (p *deploycmd) deploySingle(c *cli.Context, appName string, appf *common.Ap
 		if err != nil {
 			return err
 		}
-		if appf != nil {
-			if dir == wd {
-				setFuncInfoV20180708(ff, appf.Name)
-			}
+		if isAppYAML && dir == wd {
+			// TODO(reed): why are we setting this any differently than anything else? this is magical.
+			// for app.yaml that don't provide a func.yaml with a name, we use '$PWD-root' instead of '$PWD'
+			// for the name of the function - this is really weird. remove?
+			setFuncInfoV20180708(ff, app.Name)
```

Comment:
i find this behavior really weird, see https://github.com/fnproject/cli/pull/533/files#diff-1b0a895253821e350d6b12ecde4d6975L318 - do you think we can / should remove it and just name $PWD like everything else that doesn't provide a name in the func.yaml? this really tripped me up.

Reply:
this comment is inaccurate, it's actually `$appname-root` instead of `$PWD` that's happening - double trippy

Reply:
I agree, but it needn't be part of this PR.

Reply:
i'm cool with that, thanks